spec/index.md

```diff
@@ -5162,26 +5167,59 @@ Only the controllers of the given canister can load a snapshot.
 S.messages = Older_messages · CallMessage M · Younger_messages
 (M.queue = Unordered) or (∀ msg ∈ Older_messages. msg.queue ≠ M.queue)
 M.callee = ic_principal
-M.method_name = 'load_snapshot'
+M.method_name = 'load_canister_snapshot'
 M.arg = candid(A)
 M.caller ∈ S.controllers[A.canister_id]
 
-New_balance = S.balances[A.canister_id] - Cycles_used
+New_state = {
+  wasm_state = S.snapshots[A.canister_id].wasm_state;
+  raw_module = S.snapshots[A.canister_id].raw_module;
+  module = parse_wasm_mod(S.snapshots[A.canister_id].raw_module;
+  chunk_store = S.snapshots[A.canister_id].chunk_store;
+  public_custom_sections = parse_public_custom_sections(S.snapshots[A.canister_id].raw_module);
+  private_custom_sections = parse_private_custom_sections(S.snapshots[A.canister_id].raw_module);
+}
+Cycles_reserved = cycles_to_reserve(S, A.canister_id, S.compute_allocation[A.canister_id], S.memory_allocation[A.canister_id], S.snapshots[A.canister_id], New_state)
+New_balance = S.balances[A.canister_id] - Cycles_used - Cycles_reserved
+New_reserved_balance = S.reserved_balances[A.canister_id] + Cycles_reserved
+New_reserved_balance ≤ S.reserved_balance_limits[A.canister_id]
+
+S.canister_history[A.canister_id] = {
+  total_num_changes = N;
+  recent_changes = H;
+}
+New_canister_history = {
+  total_num_changes = N + 1;
+  recent_changes = H · {
+    timestamp_nanos = S.time[A.canister_id];
+    canister_version = S.canister_version[A.canister_id] + 1
+    origin = change_origin(M.caller, null, M.origin);
```

Comment:
`change_origin` usually has a sender_canister_version. I set parameter as null, is this correct?

Reply:
The sender_canister_version is null for ingress messages and for inter-canister calls, it is parsed from the candid argument - please see other mgmt canister methods updating history for how it's done there.

Reply:
Yes, I have checked the other methods. Basically, I wanted to ask if I should add it to the candid and pass it to this function after parsing it?